### PR TITLE
Add CLI commands for modifying plugin configs

### PR DESCRIPTION
**Pull request by foxydevloper**
This PR adds the following commands:
`omegga get-config <pluginName> [configName]` - Gets a config for a plugin. If `configName` is omitted, returns all config values.
`omegga set-config <pluginName> [configName] [configValue]` - Sets a config for a plugin. If `configValue` is omitted, the config will be reset. If `configName` is omitted, the entire plugin config will be reset.

These commands allow you to alter plugin configuration using Omegga through command line.
This currently supports the config types boolean, number, string, and enum.
`<pluginName>` must be the name of the plugin directory, it does not use the name of the plugin the documentation.

### CODE REVIEW

File: src/cli/plugin.ts [hunk 1]
```diff
@@ -703,4 +704,129 @@ async function init() {
   log('Initialized', type.yellow, 'plugin', `${name}`.cyan, 'successfully!');
 }
 
-export default { install, update, check, init };
+function pluginLoaderFactory() {
```
Comment (Meshiest) [inline src/cli/plugin.ts:707]: You should be able to make the `omegga` in the PluginLoader constructor optional:

```ts
  constructor(pluginsPath: string, omegga?: Omegga) {
    this.path = pluginsPath;
    this.omegga = omegga;
    this.store = Datastore.create({
      filename: path.join(omegga.dataPath, soft.PLUGIN_STORE),
      autoload: true,
    });
    this.store.persistence.setAutocompactionInterval(1000 * 60 * 5);
    this.formats = [];
    this.plugins = [];

    // soon to be deprecated !help
    omegga?.on('chatcmd:help', this.showHelp.bind(this));

    // use /plugins to get help
    omegga?.on('cmd:plugins', this.showHelp.bind(this));
  }
```

also you should be able to move the body of the .map.filter.map in PluginLoader.scan to a separate method called `scanPlugin(path: string)`

this line can be moved from `src/omegga/server.ts` to the PluginLoader's constructor:
    `this.pluginLoader.loadFormats(join(__dirname, 'plugin'));`

File: src/main.ts [hunk 2]
```diff
@@ -336,4 +332,58 @@ program
     pluginUtil.init();
   });
 
+program
```
Comment (Meshiest) [inline src/main.ts:335]: please also add a `reset-config [name]` to reset all or one of the config fields

**Review by Meshiest — CHANGES_REQUESTED** [review]

File: src/cli/plugin.ts (hunk 1 above)
Comment (Meshiest) [reply src/cli/plugin.ts:707]: may need to replace `pluginsPath` with just `workDir` and make these changes:

`this.path = path.join(workDir, soft.PLUGIN_PATH)`

`filename: path.join(workDir, soft.DATA_PATH, soft.PLUGIN_STORE)`

File: src/main.ts [hunk 3]
```diff
@@ -336,4 +332,97 @@ program
...
+      process.exit(1);
+    }
+    const { verbose, yes } = program.opts();
+    Logger.VERBOSE = verbose;
+
+    pluginUtil.resetAllConfigs(pluginName, yes);
+  });
+
+program
+  .command('reset-config <pluginName> <configName>')
```
Comment (Meshiest) [inline src/main.ts:410]: should be `[configName]`

<required>

[optional]

File: src/omegga/plugin.ts [hunk 4]
```diff
@@ -280,22 +290,28 @@ export class PluginLoader {
...
+      filename: path.join(workDir, soft.DATA_PATH, soft.PLUGIN_STORE),
       autoload: true,
     });
     this.store.persistence.setAutocompactionInterval(1000 * 60 * 5);
     this.formats = [];
     this.plugins = [];
 
-    // soon to be deprecated !help
-    omegga.on('chatcmd:help', this.showHelp.bind(this));
+    Logger.verbose('Creating loading plugins');
```
Comment (Meshiest) [inline src/omegga/plugin.ts:304]: this was a typo in the original command - should just be `Loading plugin formats`

**Review by Meshiest — CHANGES_REQUESTED** [review]

**Comment by Meshiest** [comment]
still waiting on the other changes

**Review by Meshiest — CHANGES_REQUESTED** [review]

File: src/cli/plugin.ts [hunk 5]
```diff
@@ -703,4 +704,131 @@ async function init() {
...
+  };
+
+  // We don't need anything except the plugin loader.
+  const options = {
+    noauth: true,
+    noplugin: false,
+    noweb: true,
+    debug: false,
+  };
+  const omegga = new Omegga(workDir, config, options);
```
Comment (Meshiest) [inline src/cli/plugin.ts]: we shouldn't need omegga anymore for plugin loaders

**Review by Meshiest — APPROVED** [review]

**Comment by foxydevloper** [comment]
done

**Pull request merged**